### PR TITLE
SC-12 # `bm server serve` prototype, no tests

### PR DESCRIPTION
**Pull request by jokeyrhyme**
### Added
- SC-12: experimental implementation of `bm server serve`
  - provides a local development server that can be used for testing

### CODE REVIEW

File: lib/serve.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  options = options || {}
+  const server = new Hapi.Server()
+  server.connection({ port: options.port })
+
+  server.route({
+    config: {
+      cors: true // development only, not recommended in production
+    },
+    handler (request, reply) {
+      const api = getAPI(options.cwd, request.params.api)
```
Comment (simonmarklar) [inline lib/serve.js:50]: I dont know Hapijs that well, but i'm wondering what happens with `request.params.api`  when you have a url like /simon/i%20am%20so%20encoded/   ??

File: commands/serve.js [hunk 2]
```diff
@@ -1,4 +1,23 @@
 'use strict'
 
+const serve = require('../lib/serve.js')
+
 module.exports = function (input, flags, logger, options) {
+  const cwd = options.cwd
+  return serve.startServer({
+    cwd,
+    port: 3000
```
Comment (simonmarklar) [inline commands/serve.js:9]: i assume that you're going to make the port configurable?

**Review by simonmarklar — COMMENTED** [review]
a couple of questions

File: commands/serve.js (hunk 2 above)
Comment (jokeyrhyme) [reply commands/serve.js:9]: There'll probably be a `--port` flag or something later. I've also contemplated automatically finding and binding to a free port.

File: lib/serve.js (hunk 1 above)
Comment (jokeyrhyme) [reply lib/serve.js:50]: Just tested this out. It automatically gets decoded:

`request.params.api === "i am so encoded"`

I will consider sanitising inputs as we stabilise this approach. This is just a local development server, so it didn't seem too important to start with. How this actually executes in AWS Lambda will probably be very different.

Comment (simonmarklar) [reply lib/serve.js:50]: cool, i figured hapi probably did something like that. its would be so annoying if you had to code that stuff up yourself for every project using hapi

File: commands/serve.js (hunk 2 above)
Comment (simonmarklar) [reply commands/serve.js:9]: sweet

**Comment by simonmarklar** [comment]
LGTM :+1:

**Pull request merged**